test/mem.test.js

```diff
@@ -888,6 +880,199 @@ describe('additional mem-store tests', () => {
+        .data$({ created_at: elvis_bday })
+        .save$((err, out) => err ? reject(err) : resolve(out))
+    }))
+
+    beforeEach(() => new Promise((resolve, reject) => {
+      seneca.make('products')
+        .data$({ created_at: millenium })
+        .save$((err, out) => err ? reject(err) : resolve(out))
+    }))
+
+
+    it('can query by date', (fin) => {
+      return seneca.make('products')
+        .load$({ created_at: makeDateSimilar(millenium) }, (err, out) => {
+          if (err) {
+            return fin(err)
+          }
+
+          expect(out).to.contain({
+            created_at: millenium
+          })
+
+          return fin()
+        })
+    })
+
+    it('can query by date', (fin) => {
+      return seneca.make('products')
+        .load$({ created_at: makeDateSimilar(elvis_bday) }, (err, out) => {
+          if (err) {
+            return fin(err)
+          }
+
+          expect(out).to.contain({
+            created_at: elvis_bday
+          })
+
+          return fin()
+        })
+    })
+
+    it('fails when trying to compare a date field to anything else', (fin) => {
+      return seneca.make('products')
+        .load$({ created_at: 123 }, (err, out) => {
+          if (err) {
+            return fin(err)
+          }
+
+          expect(out).to.equal(null)
+
+          return fin()
+        })
+    })
+  })
+
+  describe('#list by date', () => {
+    let seneca
+    
+    beforeEach(async () => {
+      seneca = makeSenecaForTest()
+    })
+
+
+    const millenium = new Date(2000, 0, 1)
+    const elvis_bday = new Date(1935, 0, 8)
+    const makeDateSimilar = similar_to => new Date(similar_to)
+
+    beforeEach(() => new Promise((resolve, reject) => {
+      seneca.make('products')
+        .data$({ created_at: elvis_bday })
+        .save$((err, out) => err ? reject(err) : resolve(out))
+    }))
+
+    beforeEach(() => new Promise((resolve, reject) => {
+      seneca.make('products')
+        .data$({ created_at: millenium })
+        .save$((err, out) => err ? reject(err) : resolve(out))
+    }))
+
+
+    it('can query by date', (fin) => {
+      return seneca.make('products')
+        .list$({ created_at: makeDateSimilar(millenium) }, (err, out) => {
+          if (err) {
+            return fin(err)
+          }
+
+          expect(out.length).to.equal(1)
+
+          expect(out[0]).to.contain({
+            created_at: millenium
+          })
+
+          return fin()
+        })
+    })
+
+    it('can query by date', (fin) => {
+      return seneca.make('products')
+        .list$({ created_at: makeDateSimilar(elvis_bday) }, (err, out) => {
+          if (err) {
+            return fin(err)
+          }
+
+          expect(out.length).to.equal(1)
+
+          expect(out[0]).to.contain({
+            created_at: elvis_bday
+          })
+
+          return fin()
+        })
+    })
+  })
+
+  describe('#remove by date', () => {
+    let seneca
+    
+    beforeEach(async () => {
```

Comment:
convert beforeEach to a utility function and call as first line of each test
beforeEach should not be used as it makes tests harder to understand in context

Reply:
@rjrodger 
https://github.com/senecajs/seneca-mem-store/pull/105/commits/e11bb885ca36bb0cf4648561dd737417d91d77ab